analysis/preprocess/Stage1_data_cleaning.R

```diff
@@ -35,474 +26,417 @@ args <- commandArgs(trailingOnly=TRUE)
-  g <- ggplot(gather(numeric_vars), aes(value)) + 
-    geom_histogram(bins = 10) + 
-    facet_wrap(~key, scales = 'free_x')
-  print(g)
-  dev.off()
-  
-  #-------------------------------------#
-  # 4. Create the final stage 1 dataset #
-  #-------------------------------------#
-  
-  # Remove inclusion/exclusion variables from dataset
-  input <- input[ , !names(input) %in% c("start_alive", "vax_gap", "vax_mixed", "vax_prior_unknown", "prior_vax1")]
-  
-  saveRDS(input, file = file.path("output", paste0("input_",cohort_name, "_stage1",".rds")), compress = "gzip")
+  input <- input %>% filter (!is.na(index_date) & index_date <= end_date_exposure & index_date >= start_date_delta)
+  consort[nrow(consort)+1,] <- c("Inclusion criteria: Index date is not before cohort end date - will remove anyone whose eligibility date + 84 days is after study end date (only those with unknown JCVI group)",
+                                 nrow(input))
   
 }
 
-# Run function using outcome group
+#Apply outcome specific exclusions criteria
+#-------------------------------------------------#
+
+input <- input %>%
+  # Exclude individuals with a recorded diagnosis of chronic neuro-degenerative diseases prior to index date
+  # dementia
+  filter(!out_date_alzheimer_disease < index_date | is.na(out_date_alzheimer_disease)) %>%
+  filter(!out_date_vascular_dementia < index_date | is.na(out_date_vascular_dementia)) %>%
+  filter(!out_date_lewy_body_dementia < index_date | is.na(out_date_lewy_body_dementia)) %>%
+  filter(!out_date_any_dementia < index_date | is.na(out_date_any_dementia)) %>%
+  filter(!out_date_other_dementias < index_date | is.na(out_date_other_dementias)) %>%
+  filter(!out_date_unspecified_dementias < index_date | is.na(out_date_unspecified_dementias)) %>%
+  # dementia risk conditions
+  filter(!out_date_cognitive_impairment_symptoms < index_date | is.na(out_date_cognitive_impairment_symptoms)) %>%
+  # parkinson's disease
+  filter(!out_date_parkinson_disease < index_date | is.na(out_date_parkinson_disease)) %>%
+  # parkinson's risk conditions
+  filter(!out_date_restless_leg_syndrome < index_date | is.na(out_date_restless_leg_syndrome)) %>%
+  filter(!out_date_rem_sleep_disorder < index_date | is.na(out_date_rem_sleep_disorder)) %>%
+  # other neurodegenerative conditions
+  filter(!out_date_motor_neurone_disease < index_date | is.na(out_date_motor_neurone_disease)) %>%
+  filter(!out_date_multiple_sclerosis < index_date | is.na(out_date_multiple_sclerosis)) %>%
+  # neurological conditions
+  filter(!out_date_migraine < index_date | is.na(out_date_migraine))
+
+consort[nrow(consort)+1,] <- c("Neuro-degenerative specific criteria: Remove those with chornic neurodegenerative diseases prior to index date",
+                               nrow(input))
+
+# Parkinson additional exclusion criteria ‘Parkinson's disease’, we will exclude individuals if they have a record of any dementia
```

Comment:
The code as written is removing outcome events but not excluding individuals. We want to exclude these individuals altogether.

As above, I think this condition is best applied at the make_model_input stage as otherwise we have to maintain this Parkison's disease cohort throughout the pipeline along with the other outcome specific cohorts.